# Design > Sample Size

<div class="alert alert-info">Calculate the required sample size for surveys and experiments</div>

The `sample_size` function determines how many observations are needed to estimate a population mean or proportion with a desired level of precision. It accounts for confidence level, incidence rate, response rate, and optionally adjusts for finite population size.

In [1]:
import pyrsm as rsm

In [2]:
## setup pyrsm for autoreload
%reload_ext autoreload
%autoreload 2
%aimport pyrsm

# Example: Sample size for a mean

Suppose you are planning a survey to estimate the average amount customers spend per visit at a retail store. Based on prior research, you expect the standard deviation of spending to be around $10. You want the estimate to be within $2 of the true mean, with 95% confidence.

How many customers do you need to survey?

In [3]:
ss = rsm.design.sample_size("mean", err_mean=2, sd_mean=10)
ss.summary()

Sample size calculation
Calculation type     : Mean
Acceptable Error     : 2
Standard deviation   : 10
Confidence level     : 0.95
Incidence rate       : 1
Response rate        : 1
Population correction: None

Required sample size     : 97
Required contact attempts: 97


The required sample size is **97**. This means you need at least 97 survey responses to estimate average customer spending within $2 of the true mean with 95% confidence.

Note that the number of *contact attempts* equals the sample size when incidence and response rates are both 100%. In practice, you may need to contact more people to achieve the desired number of completed responses.

# Example: Sample size for a proportion

You want to estimate the proportion of customers who would recommend your product. With no prior information, you assume the proportion is 0.5 (the most conservative estimate). You want the margin of error to be within 10 percentage points (0.1).

In [4]:
ss_prop = rsm.design.sample_size("proportion", err_prop=0.1, p_prop=0.5)
ss_prop.summary()

Sample size calculation
Calculation type     : Proportion
Acceptable Error     : 0.1
Proportion           : 0.5
Confidence level     : 0.95
Incidence rate       : 1
Response rate        : 1
Population correction: None

Required sample size     : 97
Required contact attempts: 97


If you have a prior estimate that the proportion is closer to 0.3, the required sample size decreases because there is less variability:

In [5]:
ss_prop2 = rsm.design.sample_size("proportion", err_prop=0.1, p_prop=0.3)
ss_prop2.summary()

Sample size calculation
Calculation type     : Proportion
Acceptable Error     : 0.1
Proportion           : 0.3
Confidence level     : 0.95
Incidence rate       : 1
Response rate        : 1
Population correction: None

Required sample size     : 81
Required contact attempts: 81


# Accounting for incidence and response rates

In real-world surveys, not everyone you contact will be eligible (incidence rate) or will respond (response rate). If you expect only 50% of contacts to be eligible and only 50% of eligible contacts to respond, the number of contact attempts increases substantially:

In [6]:
ss_ir = rsm.design.sample_size("mean", err_mean=2, sd_mean=10, incidence=0.5, response=0.5)
ss_ir.summary()

Sample size calculation
Calculation type     : Mean
Acceptable Error     : 2
Standard deviation   : 10
Confidence level     : 0.95
Incidence rate       : 0.5
Response rate        : 0.5
Population correction: None

Required sample size     : 97
Required contact attempts: 388


The required sample size is still 97, but you now need **388 contact attempts** to achieve that sample size (97 / 0.5 / 0.5 = 388).

# Population correction

When your population is relatively small, you can apply a finite population correction. This reduces the required sample size because sampling a larger fraction of the population provides more precise estimates:

In [ ]:
ss_pop = rsm.design.sample_size("mean", err_mean=2, sd_mean=10, pop_correction=True, pop_size=500)
ss_pop.summary()

Sample size calculation
Calculation type     : Mean
Acceptable Error     : 2
Standard deviation   : 10
Confidence level     : 0.95
Incidence rate       : 1
Response rate        : 1
Population correction: Yes
Population size      : 500

Required sample size     : 81
Required contact attempts: 81


With a population of 500, the required sample size is smaller than the 97 needed without population correction. The correction factor becomes more impactful as the sample becomes a larger fraction of the population.

© Vincent Nijs (2026)